### PR TITLE
Audio Sfx (code_801A5BD0 and selected sfx functions in code_8019AF00.c) OK and Documented

### PR DESCRIPTION
**Pull request by engineer124**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

The next audio file, this one primarily focusing on the high level processing of sfx's specifically.

### CODE REVIEW

File: src/code/audio/code_801A5BD0.c
Comment (EllipticEllipsis) [inline src/code/audio/code_801A5BD0.c:354]: I'm not sure what the purpose of this is, but since this is the only place these are used, you can surely name them a bit more descriptively.

File: src/code/audio/code_801A5BD0.c
Comment (EllipticEllipsis) [inline src/code/audio/code_801A5BD0.c:662]: Hmm, I see that "by" makes more sense when you have all these laid out like this... I wish there was a better way to say "with specified X (and Y).

File: src/code/audio/code_801A5BD0.c
Comment (EllipticEllipsis) [inline src/code/audio/code_801A5BD0.c:697]: I'd prefer if we didn't call a function "Unused" unless there's literally no other way to distinguish it from a used one.

**Review by EllipticEllipsis — COMMENTED** [review]
Nice job on this, although certain parts look a bit WIP still. I have a few comments for now, quite a lot is still rather over my head.

File: src/code/audio/code_801A5BD0.c
Comment (engineer124) [reply src/code/audio/code_801A5BD0.c:389]: Notice how this for-loop goes from:

`for (i = 0; i < numChannels; i++)`

So instead of using the "break" function in c, they just set i to be the end value numChannels to force the for-loop to end, which is basically equivalent to `break;`, albeit in a more hacky way

File: src/code/audio/code_801A5BD0.c
Comment (engineer124) [reply src/code/audio/code_801A5BD0.c:662]: `By` -> `Using`?        `Audio_StopSfxUsingPosAndId`?
`By` -> `With`           `Audio_StopSfxWithPosAndId`?
`By` -> `Through`     `Audio_StopSfxThroughPosAndId`?

**Comment by engineer124** [comment]
For reference, I decided to change the following names, although feedback is still welcome:
`gSfxPosScreenCenter` -> `gSfxDefaultPosScreenCenter`
`gSfxDefaultVolOrFreq` -> `gSfxVolOrFreqDefaultVal`
`gSfxDefaultReverbAddNone` -> `gSfxReverbAddNone`

File: src/code/audio/code_801A5BD0.c
Comment (EllipticEllipsis) [reply src/code/audio/code_801A5BD0.c:699]: Of those I like `With` the best, although it's still not perfect. `Through` doesn't sound right to me, and `Using` suggests that the properties are more involved than just being a category type thing.

File: src/code/audio/code_801A5BD0.c
Comment (EllipticEllipsis) [inline src/code/audio/code_801A5BD0.c:20]: Is it known what these are?

Comment (engineer124) [reply src/code/audio/code_801A5BD0.c:20]: It's true-false data for each of the 7 sound-banks:
https://github.com/zeldaret/mm/blob/ad42edb0340ba5e0e2f6e318b58fcb0af61312d1/src/code/audio/code_8019AF00.c#L156-L162

In OoT, there's only 1, not 2, and it's called `gIsLargeSoundBank`, but I don't think that name is accurate to begin with, so I refrained naming these.
https://github.com/zeldaret/oot/blob/32e66c2da8c2f9816773e1b2a0bc92804e9d20de/src/code/code_800EC960.c#L87

Nonetheless, these externs should be in variables.h I just noticed

File: src/code/audio/code_801A5BD0.c
Comment (EllipticEllipsis) [reply src/code/audio/code_801A5BD0.c]: Oh dear. Hmm, I think it would be better to comment on what it's doing in English rather than C... `// Makes this pass the last one, essentially "break;"`?

Comment (engineer124) [reply src/code/audio/code_801A5BD0.c]: In my latest push, I did expand on this comment. Let me know if it needs rephrasing again

File: src/code/audio/code_801A5BD0.c
Comment (hensldm) [inline src/code/audio/code_801A5BD0.c:548]: I'll defer to your expertise, but looks like in OOT the last argument to this function is called `data`, so perhaps `data` for the stack variable instead of `temp`?

**Review by hensldm — APPROVED** [review]
One possible naming suggestion.

File: src/code/audio/code_801A5BD0.c
Comment (engineer124) [reply src/code/audio/code_801A5BD0.c:548]: Yeah, I'm fine with `data` for now. This is sending `temp`/`data` to io-port 5 in the sfx sequence, so we'll find a better name for it eventually once we properly have sequence 0 understood

File: src/code/audio/code_801A5BD0.c
Comment (EllipticEllipsis) [inline src/code/audio/code_801A5BD0.c:423]: How about
```suggestion
                        i = numChannels; // make this last pass through loop, here equivalent to "break"
```
mainly because this is *not* equivalent to break if you add stuff later on.

File: src/code/audio/code_801A5BD0.c
Comment (AngheloAlf) [inline src/code/audio/code_801A5BD0.c:209]: Would it be better this way?
```suggestion
    u8 index;
    SfxRequest* req = &sSfxRequests[sSfxRequestReadIndex];
    SfxBankEntry* entry;
    SfxParams* sfxParams;
    s32 bankId;
    u8 evictImportance;
    u8 evictIndex = 0x80;

    if (req->sfxId == 0) {
```

**Comment by engineer124** [comment]
I'll be drafting this until PlayState is renamed https://github.com/zeldaret/mm/pull/835

**Comment by AngheloAlf** [comment]
This PR can now be undrafted or is it waiting for something else?

**Comment by engineer124** [comment]
Sorry, should have mentioned this PR is on hold for https://github.com/zeldaret/mm/pull/927

**Comment by engineer124** [comment]
This PR is now ready for review again. Documentation has had an overhaul due to new and improved knowledge. Also, a few functions from `code_8019AF00.c` are also part of the audio sfx system and are now included in this PR, as those functions closely tie in with `code_801A5BD0.c`

File: src/code/audio/code_8019AF00.c
Comment (hensldm) [inline src/code/audio/code_8019AF00.c:157]: Hmm, I'm not really a big fan of the name, though don't really know enough to suggest a better one. I see one usage down below. Is that the only one or are there others?

**Review by hensldm — CHANGES_REQUESTED** [review]
Some thoughts.

File: src/code/audio/code_8019AF00.c
Comment (engineer124) [reply src/code/audio/code_8019AF00.c:3919]: I've definitely been conscious of when to include `Sfx` in the function name and when not to. In this particular case, I included it because the struct itself is called `SfxChannelState`, but `AudioSfx_ResetChannelState` could work too

File: src/code/audio/code_8019AF00.c
Comment (engineer124) [reply src/code/audio/code_8019AF00.c:157]: Oh right. OoT has this variable called `gIsLargeSfxBank` which is a much cleaner name, but it took me forever to figure out what it means. It's specifically because the sequence needs to store the ID in a `u8` value, but can only hold `0xFF` values. So if the number of sfx is greater than 255, then the sequence needs two `u8` to store the lower and upper bits. Perhaps this is better explained in comments instead of a verbose temp name, and I can just stick with `gIsLargeSfxBank`

Comment (hensldm) [reply src/code/audio/code_8019AF00.c:157]: Personally I like `gIsLargeSfxBank` with the comment. What you have just seems a bit too on the nose for me.

File: tools/namefixer.py
Comment (AngheloAlf) [inline tools/namefixer.py:262]: Could you add `Audio_PlaySfxGeneral` -> `AudioSfx_PlaySfx`?

File: tools/namefixer.py
Comment (AngheloAlf) [inline tools/namefixer.py:151]: Same for `Audio_StopSfxByPos` -> `AudioSfx_StopByPos`

File: include/z64audio.h
Comment (engineer124) [reply include/z64audio.h:36]: `SOUNDMODE_SURROUND` seems to be external and used in high-level processing, while `SOUNDMODE_SURROUND_ALT` seems to be internal and used in low-level processing. But I'm pretty sure it's the same enum. It's kinda weird

Comment (AngheloAlf) [reply include/z64audio.h]: Yeah I don't doubt it is the same enum, I was just curious about what the difference was

**Review by AngheloAlf — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**
